`blendphase` raises when using LLVM variant

## Summary

The `blendphase` plugin's `sample()` method raises when used with a JIT variant.

## System configuration

```
System information:

  OS: Ubuntu 22.04.1 LTS
  CPU: AMD Ryzen 9 5900X 12-Core Processor
  GPU: NVIDIA GeForce GT 730
  Python: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:18) [GCC 10.3.0]
  NVidia driver: 470.161.03
  LLVM: 14.0.0

  Dr.Jit: 0.4.0
  Mitsuba: 3.2.0
     Is custom build? True
     Compiled with: Clang 11.1.0
     Variants:
        scalar_mono
        scalar_mono_double
        scalar_rgb
        scalar_spectral
        llvm_mono_double
        llvm_ad_rgb
```

## Details

I pinpointed the error to [this line in `blendphase.cpp`](https://github.com/mitsuba-renderer/mitsuba3/blob/8ee2606087f210c0af8c486d66225d1b7e3d2729/src/phase/blendphase.cpp#L134) (134).
## Steps to reproduce

1. At [this line of the test file](https://github.com/mitsuba-renderer/mitsuba3/blob/d6b2b2b0f1fdcbdd81fd027543c098cf27247544/src/phase/tests/test_blendphase.py#L64) (64), replace `variant_scalar_rgb` with `variants_all_rgb`
2. Run `pytest src/phase/tests/test_blendphase.py -k "test03_sample_all"`

Expected error:
```
_____________________________________________ test03_sample_all[llvm_ad_rgb] _____________________________________________

variants_all_rgb = 'llvm_ad_rgb'

    def test03_sample_all(variants_all_rgb):
        weight = 0.2
        g = 0.2
    
        phase = mi.load_dict(
            {
                "type": "blendphase",
                "phase1": {"type": "isotropic"},
                "phase2": {"type": "hg", "g": g},
                "weight": weight,
            }
        )
    
        mei = mi.MediumInteraction3f()
        mei.t = 0.1
        mei.p = [0, 0, 0]
        mei.sh_frame = mi.Frame3f([0, 0, 1])
        mei.wi = [0, 0, 1]
    
        ctx = mi.PhaseFunctionContext(None)
    
        # Sample using two different values of 'sample1' and make sure correct
        # components are chosen.
    
        # -- Sample above weight: first component (isotropic) is selected
        expected_a = dr.inv_four_pi
>       wo_a, pdf_a = phase.sample(ctx, mei, 0.3, [0.5, 0.5])
E       RuntimeError: jit_var_select(r52, r102, r0): operands have incompatible sizes! (sizes: 1, 1, 0)

src/phase/tests/test_blendphase.py:91: RuntimeError
```

--- a/src/phase/blendphase.cpp
+++ b/src/phase/blendphase.cpp
@@ -122,8 +122,8 @@ public:
             return { wo, pdf };
         }
 
-        Vector3f wo;
-        Float pdf;
+        Vector3f wo = dr::zeros<Vector3f>();
+        Float pdf = dr::zeros<Float>();
 
         Mask m0 = active && sample1 > weight,
              m1 = active && sample1 <= weight;

--- a/src/phase/tests/test_blendphase.py
+++ b/src/phase/tests/test_blendphase.py
@@ -61,7 +61,7 @@ def test02_eval_all(variant_scalar_rgb):
     assert dr.allclose(value, expected)
 
 
-def test03_sample_all(variant_scalar_rgb):
+def test03_sample_all(variants_all_rgb):
     weight = 0.2
     g = 0.2
 

